freeipa/src/main/java/com/sequenceiq/freeipa/service/recipe/FreeIpaRecipeService.java

```diff
@@ -0,0 +1,70 @@
+package com.sequenceiq.freeipa.service.recipe;
+
+import static com.sequenceiq.authorization.resource.AuthorizationResourceType.RECIPE;
+
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import javax.inject.Inject;
+
+import org.apache.commons.codec.binary.Base64;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+
+import com.sequenceiq.authorization.resource.AuthorizationResourceType;
+import com.sequenceiq.authorization.service.AuthorizationResourceCrnListProvider;
+import com.sequenceiq.cloudbreak.api.endpoint.v4.common.CompactViewV4Response;
+import com.sequenceiq.cloudbreak.api.endpoint.v4.recipes.RecipeV4Endpoint;
+import com.sequenceiq.cloudbreak.api.endpoint.v4.recipes.requests.RecipeV4Request;
+import com.sequenceiq.cloudbreak.api.endpoint.v4.recipes.requests.RecipeV4Type;
+import com.sequenceiq.cloudbreak.api.endpoint.v4.recipes.responses.RecipeViewV4Responses;
+import com.sequenceiq.cloudbreak.common.model.recipe.RecipeType;
+import com.sequenceiq.cloudbreak.orchestrator.model.RecipeModel;
+import com.sequenceiq.cloudbreak.service.CloudbreakRuntimeException;
+
+@Service
+public class FreeIpaRecipeService implements AuthorizationResourceCrnListProvider {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(FreeIpaRecipeService.class);
+
+    @Inject
+    private RecipeV4Endpoint recipeV4Endpoint;
+
+    @Override
+    public List<String> getResourceCrnListByResourceNameList(List<String> resourceNames) {
+        RecipeViewV4Responses recipes = recipeV4Endpoint.list(0L);
+        List<String> recipeNamesFromCore = recipes.getResponses().stream().map(CompactViewV4Response::getName).collect(Collectors.toList());
+        List<String> recipesNotFound = resourceNames.stream().filter(recipeName -> !recipeNamesFromCore.contains(recipeName)).collect(Collectors.toList());
+        if (recipesNotFound.size() > 0) {
+            throw new CloudbreakRuntimeException("Following recipes does not exist: " + recipesNotFound);
+        }
```

Comment:
similarly to env service, couldn;'t this be a notfound exception? and a separate method. this feels like code duplication, if we have the same in core, maybe it should be refactored into something common like we have environment-common module

Reply:
Unfortunately, there is no common module where core api is accessible and I need to call recipeV4Endpoint.list(0L);
I tried to put it into some common place, but I failed. I can create a new module core-common like environment-common, but I don't really want to create it for 1 method. (okay maybe there will be more later). Maybe you have some better solution in mind.

Reply:
I don't say it must be part of this PR, but if the 2 or more services share most of the code then we should really think about a recipe-common or similar module to extract common logic. I don't have better idea than this at the moment. so I resolve this for now

Reply:
Okay, I created a common module. Fortunately it is not as a bid deal as I thought :) Thanks for the idea!